src/state/position-permissions/hooks.ts

```diff
@@ -1,11 +1,42 @@
 import some from 'lodash/some';
 import isEqual from 'lodash/isEqual';
 import sortBy from 'lodash/sortBy';
+import useWeb3Service from 'hooks/useWeb3Service';
 import { useMemo } from 'react';
 import { useAppSelector } from 'state/hooks';
+import { FullPermission, Permission } from 'types';
 import { RootState } from '../index';
 
-export function usePositionPermissions(positionId: string) {
+export type PositionPermissions = {
+  isOwner: boolean;
+  INCREASE?: string;
+  REDUCE?: string;
+  WITHDRAW?: string;
+  TERMINATE?: string;
+};
+
+export const createPermissionsObject = (permissions: Permission[] = [], isOwner = false) => {
+  const permissionsToUse = isOwner ? ['WITHDRAW', 'TERMINATE', 'REDUCE', 'INCREASE'] : permissions;
+  return permissionsToUse.reduce((acc, curr) => ({ ...acc, [curr]: true }), { isOwner });
+};
+
+export function mergeCompanionPermissions(
```

Comment:
The goal is getting a permission object that only has ```true``` for permissions that are both present in the ```accountsPermissions``` and the ```companionPermissions``` so for example, if you send it
```
// companion
{ isOwner: false, TERMINATE: true, WITHDRAW: true}

// account

{ isOwner: false, TERMINATE: true, REDUCE: true}
```

You will receive

```
{ isOwner: false, TERMINATE: true }
```

Which is the only overlap in truthy permissions. This merging of permissions is used in cases where there is yield of protocol tokens, to reduce the amount of inline conditionals ```hasYield ? accountPermissions.TERMINATE && companionPermissions.TERMINATE : accountPermission.TERMINATE &&...``` . So I conditionally merged user and companion permissions in some components, based on yield or protocol token (like in ```src/home/positions-list/components/position-controls/index.tsx:198``` ) to avoid redundant conditional code. WDYT?

If the user is owner it just returns their permissions (which should aways be all of them) because the companion is not needed.